commands/config_test.go

```diff
@@ -0,0 +1,392 @@
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Set an int config value for a given key", func() {
+		printer.Clean()
+		args := []string{"SqlSettings.MaxIdleConns", "20"}
+		defaultConfig := &model.Config{}
+		defaultConfig.SetDefaults()
+		inputConfig := &model.Config{}
+		inputConfig.SetDefaults()
+		changedValue := 20
+		inputConfig.SqlSettings.MaxIdleConns = &changedValue
+
+		s.client.
+			EXPECT().
+			GetConfig().
+			Return(defaultConfig, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			UpdateConfig(inputConfig).
+			Return(inputConfig, &model.Response{Error: nil}).
+			Times(1)
+
+		err := configSetCmdF(s.client, &cobra.Command{}, args)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0].(string), "Value changed successfully")
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Set a boolean config value for a given key", func() {
+		printer.Clean()
+		args := []string{"SqlSettings.Trace", "true"}
+		defaultConfig := &model.Config{}
+		defaultConfig.SetDefaults()
+		inputConfig := &model.Config{}
+		inputConfig.SetDefaults()
+		changedValue := true
+		inputConfig.SqlSettings.Trace = &changedValue
+
+		s.client.
+			EXPECT().
+			GetConfig().
+			Return(defaultConfig, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			UpdateConfig(inputConfig).
+			Return(inputConfig, &model.Response{Error: nil}).
+			Times(1)
+
+		err := configSetCmdF(s.client, &cobra.Command{}, args)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0].(string), "Value changed successfully")
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Set a slice of string config value for a given key", func() {
+		printer.Clean()
+		args := []string{"SqlSettings.DataSourceReplicas", "test1", "test2"}
+		defaultConfig := &model.Config{}
+		defaultConfig.SetDefaults()
+		inputConfig := &model.Config{}
+		inputConfig.SetDefaults()
+		inputConfig.SqlSettings.DataSourceReplicas = []string{"test1", "test2"}
+
+		s.client.
+			EXPECT().
+			GetConfig().
+			Return(defaultConfig, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			UpdateConfig(inputConfig).
+			Return(inputConfig, &model.Response{Error: nil}).
+			Times(1)
+
+		err := configSetCmdF(s.client, &cobra.Command{}, args)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0].(string), "Value changed successfully")
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Get error if the key doesn't exists", func() {
+		printer.Clean()
+		defaultConfig := &model.Config{}
+		defaultConfig.SetDefaults()
+		args := []string{"SqlSettings.WrongKey", "test1"}
+		inputConfig := &model.Config{}
+		inputConfig.SetDefaults()
+
+		s.client.
+			EXPECT().
+			GetConfig().
+			Return(defaultConfig, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
```

Comment:
Mocking this interaction is not needed, as the `gomock` library will fail on any interaction with the client that is not mocked already